File: backend/src/main/scala/domains/problem/objects/response/ProblemDataFileListResponse.scala
package domains.problem.objects.response

import domains.problem.objects.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

/** 旧版题目数据文件列表响应；只包含根级文件名列表。 */
final case class ProblemDataFileListResponse(items: List[ProblemDataFilename])

/** ProblemDataFileListResponse 的 JSON 编解码器。 */
object ProblemDataFileListResponse:
  given Encoder[ProblemDataFileListResponse] = deriveEncoder[ProblemDataFileListResponse]
  given Decoder[ProblemDataFileListResponse] = deriveDecoder[ProblemDataFileListResponse]

File: frontend/src/objects/problem/response/ProblemDataFileListResponse.ts
import type { ProblemDataFilename } from '@/objects/problem/ProblemDataFilename'

/** 题目数据文件名列表响应；仅返回当前目录/数据包可见的文件名集合。 */
export type ProblemDataFileListResponse = {
  items: ProblemDataFilename[]
}
